lib/src/configuration.dart

```diff
@@ -379,4 +383,69 @@ class SyncClientResetErrorHandler {
 }
 
 /// A client reset strategy where the user needs to fully take care of a client reset.
-typedef ManualSyncClientResetHandler = SyncClientResetErrorHandler;
+///
+/// If you set [ManualSyncClientResetHandler] callback as `syncClientResetErrorHandler` argument of [Configuration.flexibleSync],
+/// that will enable full control of moving unsynced changes to synced realm.
+/// {@category Sync}
+class ManualSyncClientResetHandler extends SyncClientResetErrorHandler {
+  const ManualSyncClientResetHandler(super.callback);
+}
+
+/// A client reset strategy where all the not yet synchronized data is automatically
+/// discarded and a fresh copy of the synchronized Realm is obtained.
+///
+/// If you set [DiscardLocalSyncClientResetHandler] callback as `syncClientResetErrorHandler` argument of [Configuration.flexibleSync],
+/// the local Realm will be discarded and replaced with the server side Realm.
+/// All local changes will be lost.
+/// {@category Sync}
+class DiscardLocalSyncClientResetHandler extends SyncClientResetErrorHandler {
```

Comment:
```suggestion
class DiscardUnsyncedChangesHandler extends SyncClientResetErrorHandler {
```

Reply:
DiscardUnsyncedChangesClientResetErrorHandler? . will be  DiscardUnsyncedChangesClientResetExceptionHandler?